scene/gui/line_edit.cpp

```diff
@@ -619,6 +619,44 @@ void LineEdit::gui_input(const Ref<InputEvent> &p_event) {
 		return;
 	}
 
+	// Open context menu.
+	if (context_menu_enabled) {
+		if (k->is_action("ui_menu", true)) {
+			_update_context_menu();
+			Point2 pos = Point2(get_caret_pixel_pos().x, (get_size().y + theme_cache.font->get_height(theme_cache.font_size)) / 2);
+			menu->set_position(get_screen_position() + pos);
+			menu->reset_size();
+			menu->popup();
+			menu->grab_focus();
+
+			accept_event();
+			return;
+		}
+	}
+
+	if (is_shortcut_keys_enabled()) {
+		// Cut
+		if (k->is_action("ui_copy", true)) {
+			copy_text();
+			accept_event();
+			return;
+		}
+
+		// Select all
+		if (k->is_action("ui_text_select_all", true)) {
+			select();
+			accept_event();
+			return;
+		}
+
+		// Cut
+		if (k->is_action("ui_cut", true)) {
+			cut_text();
+			accept_event();
+			return;
+		}
```

Comment:
Should not be here.

Reply:
Do you mean it should be together with `ui_paste`?

Reply:
I mean it should be after `editing` check, like before.
If LineEdit is not editable, cutting should be disabled completely, not do a copy or something else.